sklearnex/cluster/k_means.py

```diff
@@ -1,17 +1,370 @@
+
+        else:
+
+            @wrap_output_data
+            def predict(
+                self,
+                X,
+                sample_weight="deprecated" if sklearn_check_version("1.3") else None,
+            ):
+                if sklearn_check_version("1.0"):
+                    self._check_feature_names(X, reset=False)
+                if sklearn_check_version("1.2"):
+                    self._validate_params()
+                return dispatch(
+                    self,
+                    "predict",
+                    {
+                        "onedal": self.__class__._onedal_predict,
+                        "sklearn": sklearn_KMeans.predict,
+                    },
+                    X,
+                    sample_weight,
+                )
+
+        def _onedal_predict(self, X, sample_weight=None, queue=None):
+            X = self._validate_data(
+                X,
+                accept_sparse="csr",
+                reset=False,
+                dtype=[np.float64, np.float32],
+            )
+
+            if not sklearn_check_version("1.5"):
+                if (
+                    sklearn_check_version("1.3")
+                    and isinstance(sample_weight, str)
+                    and sample_weight == "deprecated"
+                ):
+                    sample_weight = None
+
+                if sklearn_check_version("1.3") and sample_weight is not None:
+                    warnings.warn(
+                        "'sample_weight' was deprecated in version 1.3 and "
+                        "will be removed in 1.5.",
+                        FutureWarning,
+                    )
+
+            if not hasattr(self, "_onedal_estimator"):
+                self._initialize_onedal_estimator()
+                self._onedal_estimator.cluster_centers_ = self.cluster_centers_
+
+            return self._onedal_estimator.predict(X, queue=queue)
+
+        def _onedal_supported(self, method_name, *data):
+            if method_name == "fit":
+                return self._onedal_fit_supported(method_name, *data)
+            if method_name == "predict":
+                return self._onedal_predict_supported(method_name, *data)
+            raise RuntimeError(
+                f"Unknown method {method_name} in {self.__class__.__name__}"
+            )
+
+        def _onedal_gpu_supported(self, method_name, *data):
+            return self._onedal_supported(method_name, *data)
+
+        def _onedal_cpu_supported(self, method_name, *data):
+            return self._onedal_supported(method_name, *data)
+
+        @wrap_output_data
+        def fit_transform(self, X, y=None, sample_weight=None):
+            return self.fit(X, sample_weight=sample_weight)._transform(X)
+
+        @wrap_output_data
+        def transform(self, X):
+            check_is_fitted(self)
+
+            X = self._check_test_data(X)
+            return self._transform(X)
+
+        score = support_usm_ndarray()(sklearn_KMeans.score)
```

Comment:
should implement our own score method, since we need to access our own predict. This has implications with onedal_supported etc,